runtime/tr.source/trj9/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -18728,15 +18728,13 @@ J9::Z::TreeEvaluator::pd2lVariableEvaluator(TR::Node* node, TR::CodeGenerator* c
       regPair = cg->allocateConsecutiveRegisterPair(LReg, HReg);
       }
 
-   // byte-length = precision/2 + 1
    TR::Register* callAddrReg = cg->evaluate(pdAddressNode);
    TR::Register* precisionReg = cg->evaluate(pdOpNode->getChild(2));
+   TR::Register* lengthReg = cg->allocateRegister();
    TR_ASSERT(precisionReg && (precisionReg->getKind() == TR_GPR), "precision should be a 32bit GPR");
 
-   TR::Register* lengthReg = cg->allocateRegister();
-   generateRRInstruction(cg, TR::InstOpCode::LR, node, lengthReg, precisionReg);
-   generateRSInstruction(cg, TR::InstOpCode::SRA, pdOpNode, lengthReg, lengthReg, 0x1, NULL);
-   generateRIInstruction(cg, TR::InstOpCode::AGHI, node, lengthReg, 0x1);
+   // byte-length = precision/2 + 1. Note that the length codes of all instructions are (byte-length-1)
+   generateRSInstruction(cg, TR::InstOpCode::SRAK, pdOpNode, lengthReg, precisionReg, 0x1, NULL);
```

Comment:
I think SRAK is only available on z196 and newer.  Can you double check if we can unconditionally use this?

Reply:
Nice catch. Fixed SRAK.